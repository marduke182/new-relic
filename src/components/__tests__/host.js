import hostComponent from '../host';

const createApp = (apdex) => ({ name: `App${apdex}`, apdex: apdex, version: 1, contributors: [], host: [] })

const host1 = { name: '12345.host1.com' };

function createIntArray(start, end) {
  const newArray = [];

  while (start < end) {
    newArray.push(start++);
  }

  return newArray;
}

function setup({ name = 'none', apps = []} = {}) {
  const dom = hostComponent({ name, apps });
  return {
    dom,
    get apps() {
      return dom.getElementsByClassName('Host-app');
    }
  }
}

test('should render correct structure', async () => { // Check if the layout didn't change
  const host = setup({ name: host1.name, apps: [createApp(90), createApp(78)]});

  expect(host.dom.innerHTML).toMatchSnapshot();
});

test('should render first 5 apps', async () => {
  const top25apps = createIntArray(76, 100).reverse().map(createApp);
  const host = setup({ name: host1.name, apps: top25apps});

  expect(host.apps.length).toBe(5);
});
