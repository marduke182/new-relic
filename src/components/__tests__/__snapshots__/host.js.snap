// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correct structure 1`] = `
"
  <header class=\\"Host-header\\">
    <p>12345.host1.com</p>
  </header>
  <div class=\\"Host-appsContainer\\">
    <ul class=\\"Host-appsList\\">
        <li class=\\"Host-app\\" data-version=\\"1\\">
          <div class=\\"Host-appDex\\"><p>90</p></div>
          <div class=\\"Host-appName\\"><p>App90</p></div>
        </li>
        <li class=\\"Host-app\\" data-version=\\"1\\">
          <div class=\\"Host-appDex\\"><p>78</p></div>
          <div class=\\"Host-appName\\"><p>App78</p></div>
        </li>
    </ul>
  </div>
"
`;
